src/config.ts

```diff
@@ -0,0 +1,230 @@
+// Public certificate used in SAML authentication to a SPID IDP.
+const samlCert = () => {
+  return readFile(
+    process.env.SAML_CERT_PATH || "./certs/cert.pem",
+    "SAML certificate"
+  );
+};
+
+export const SAML_CERT = samlCert();
+
+// SAML settings.
+const SAML_CALLBACK_URL =
+  process.env.SAML_CALLBACK_URL ||
+  "http://italia-backend/assertionConsumerService";
+const SAML_ISSUER = process.env.SAML_ISSUER || "https://spid.agid.gov.it/cd";
+const DEFAULT_SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX = "1";
+const SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX: number = parseInt(
+  process.env.SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX ||
+    DEFAULT_SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX,
+  10
+);
+const DEFAULT_SAML_ACCEPTED_CLOCK_SKEW_MS = "-1";
+const SAML_ACCEPTED_CLOCK_SKEW_MS = parseInt(
+  process.env.SAML_ACCEPTED_CLOCK_SKEW_MS ||
+    DEFAULT_SAML_ACCEPTED_CLOCK_SKEW_MS,
+  10
+);
+const DEFAULT_SPID_AUTOLOGIN = "";
+const SPID_AUTOLOGIN = process.env.SPID_AUTOLOGIN || DEFAULT_SPID_AUTOLOGIN;
+const DEFAULT_SPID_TESTENV_URL = "https://spid-testenv2:8088";
+const SPID_TESTENV_URL =
+  process.env.SPID_TESTENV_URL || DEFAULT_SPID_TESTENV_URL;
+
+// Register the spidStrategy.
+export const IDP_METADATA_URL = getRequiredENVVar("IDP_METADATA_URL");
+export function generateSpidStrategy(): Promise<SpidStrategy> {
+  return spidStrategy(
+    SAML_KEY,
+    SAML_CERT,
+    SAML_CALLBACK_URL,
+    SAML_ISSUER,
+    SAML_ACCEPTED_CLOCK_SKEW_MS,
+    SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX,
+    SPID_AUTOLOGIN,
+    SPID_TESTENV_URL,
+    IDP_METADATA_URL
+  );
+}
+
+// Redirection urls
+const clientProfileRedirectionUrl =
+  process.env.CLIENT_REDIRECTION_URL || "/profile.html?token={token}";
+
+if (!clientProfileRedirectionUrl.includes("{token}")) {
+  log.error("CLIENT_REDIRECTION_URL must contains a {token} placeholder");
+}
+
+// Range IP allowed for notification.
+function decodeNotifyCIDR(): string & IPatternStringTag<string> {
+  const errorOrNotifyCIDR = CIDR.decode(
+    process.env.ALLOW_NOTIFY_IP_SOURCE_RANGE
+  );
+  if (isLeft(errorOrNotifyCIDR)) {
+    log.error(
+      "Missing or invalid ALLOW_NOTIFY_IP_SOURCE_RANGE environment variable: %s",
+      ReadableReporter.report(errorOrNotifyCIDR)
+    );
+    return process.exit(1);
+  } else {
+    return errorOrNotifyCIDR.value;
+  }
+}
+export const ALLOW_NOTIFY_IP_SOURCE_RANGE = decodeNotifyCIDR();
+
+// Range IP allowed for PagoPA proxy.
+function decodePagoPACIDR(): string & IPatternStringTag<string> {
+  const errorOrPagoPACIDR = CIDR.decode(
+    process.env.ALLOW_PAGOPA_IP_SOURCE_RANGE
+  );
+  if (isLeft(errorOrPagoPACIDR)) {
+    log.error(
+      "Missing or invalid ALLOW_PAGOPA_IP_SOURCE_RANGE environment variable: %s",
+      ReadableReporter.report(errorOrPagoPACIDR)
+    );
+    return process.exit(1);
+  } else {
+    return errorOrPagoPACIDR.value;
+  }
+}
+export const ALLOW_PAGOPA_IP_SOURCE_RANGE = decodePagoPACIDR();
+
+export const API_KEY = process.env.API_KEY || "";
```

Comment:
these are no sensible defaults, server should not start without these

Reply:
Fixed in 7db18a3